src/pages/index.js

```diff
@@ -67,36 +100,99 @@ function handleImageClick(evt) {
   };
   imageModal.open(imageData);
 }
+function avatarButtonClickHandler() {
+  popupAvatarForm.open();
+}
+
+function avatarFormSubmitHandler(avatarLink) {
+  const avatarUrl = avatarLink.url;
+  return api
+    .patchAvatar(avatarUrl)
+    .then(() => {
+      user.setUserAvatar(avatarUrl);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
 function editButtonClickHandler() {
-  const currentUserInfo = userInfo.getUserInfo();
-  profileFormName.value = currentUserInfo.name;
-  profileFormDesc.value = currentUserInfo.description;
+  const currentUser = user.getUserInfo();
+  profileFormName.value = currentUser.name;
+  profileFormDesc.value = currentUser.about;
   profileFormValidator.enableFormButton();
   popupProfileForm.open();
 }
 function profileFormSubmitHandler(inputValues) {
-  userInfo.setUserInfo(inputValues);
+  return api
+    .patchUserInfo(inputValues)
+    .then(user.setUserInfo(inputValues))
+    .catch((err) => {
+      console.error(err);
+    });
 }
+
 function addCardButtonClickHandler() {
   popupAddCardForm.open();
 }
-function handleAddCardFormSubmit(inputValues) {
+function addCardFormSubmitHandler(inputValues) {
   const cardData = {
     name: inputValues.title,
     link: inputValues.url,
   };
-  section.addItem(createCard(cardData));
-  popupAddCardForm.close();
-  addCardFormValidator.disableFormButton();
-  addCardForm.reset();
+
+  return api
+    .postNewCard(cardData)
+    .then((newCard) => {
+      section.addItem(createCard(newCard));
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      popupAddCardForm.close();
+      addCardFormValidator.disableFormButton();
+      addCardForm.reset();
```

Comment:
You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well